### ISSUE TITLE
Handling the "Initialization autoloaded the constants" deprecation warning

### ISSUE THREAD

**Issue by mattmenefee** [oivoodoo/devise_masquerade#99]
Currently the [README](https://github.com/oivoodoo/devise_masquerade/blob/v1.3.9/README.md) suggests adding the following configuration, presumably to the bottom of the Devise initializer:

```ruby
Devise.masquerade_param = 'masquerade'
Devise.masquerade_expires_in = 10.seconds
Devise.masquerade_key_size = 16 # size of the generate by SecureRandom.urlsafe_base64
Devise.masquerade_bypass_warden_callback = false
Devise.masquerade_routes_back = false # if true, route back to the page the user was on via redirect_back
Devise.masquerading_resource_class = AdminUser
# optional, default: masquerading_resource_class.model_name.param_key
Devise.masquerading_resource_name = :admin_user
Devise.masqueraded_resource_class = User
# optional, default: masqueraded_resource_class.model_name.param_key
Devise.masqueraded_resource_name = :user
```

However, specifying these two lines:

```ruby
Devise.masquerading_resource_class = AdminUser
Devise.masqueraded_resource_class = User
```
...will result in the following deprecation warning being thrown when the app loads:

```bash
DEPRECATION WARNING: Initialization autoloaded the constants AdminUser and User.

Being able to do this is deprecated. Autoloading during initialization is going
to be an error condition in future versions of Rails.

Reloading does not reboot the application, and therefore code executed during
initialization does not run again. So, if you reload User, for example,
the expected changes won't be reflected in that stale Class object.

These autoloaded constants have been unloaded.

In order to autoload safely at boot time, please wrap your code in a reloader
callback this way:

    Rails.application.reloader.to_prepare do
      # Autoload classes and modules needed at boot time here.
    end

That block runs when the application boots, and every time there is a reload.
For historical reasons, it may run twice, so it has to be idempotent.

Check the "Autoloading and Reloading Constants" guide to learn more about how
Rails autoloads and reloads.
```

This will prevent the app from loading completely in Rails v7.0.

To fix it, I suppose one could require the reference to those models at the top of the initializer, but I'm wondering if it would be better to specify those configuration settings as strings (while appending `_name` to the end, for clarity):

```ruby
Devise.masquerading_resource_class_name = 'AdminUser'
Devise.masqueraded_resource_class_name = 'User'
```

And then call `masquerading_resource_class.classify.constantize` where they are used elsewhere in the library? Logistically, however, I'm not sure if that would be possible. But anyways, thoughts?

Separately, would you be interested in a pull request to set up an installation generator that would append the configuration (mentioned above) to the bottom of the Devise initializer by running something like `rails generate devise_masquerade:install` after installing the gem? Other Devise gems (such as [devise_invitable](https://github.com/scambra/devise_invitable)) provide for such a thing.

**Comment by oivoodoo** [comment]
@mattmenefee Happy New Year! :) Going to read it in the new year and write you
back

**Comment by oivoodoo** [comment]
Hi @mattmenefee 

Have you tried to specify these settings :

```
Devise.masquerading_resource_class = AdminUser
Devise.masqueraded_resource_class = User
```

inside of the devise config block?

For example:

```
Devise.setup do |config|
  # The secret key used by Devise. Devise uses this key to generate
  # random tokens. Changing this key will render invalid all existing
  # confirmation, reset password and unlock tokens in the database.
  config.secret_key = "...."
  ....
  config.masquerading_resource_class = AdminUser
  config.masqueraded_resource_class = User
  ...
end
```

I am not sure about it but the gem is using  the same approach like `devise_invitable` to define the devise settings.

I don't like the idea to use `.classify.constantize` on strings because it would be a security issue probably on gem sources review on converting the strings to the modules / classes. 

if the config block works for you it means it would be lazy load for the contents. I think so.

**Comment by mattmenefee** [comment]
@oivoodoo yes, I've tried using something like `config.masquerading_resource_class = AdminUser`, but that is precisely what is resulting in the deprecation message, because the `AdminUser` constant is being referenced inside the Devise initializer before the `AdminUser` model class is has been loaded using the normal Rails autoloading process (via zeitwerk).

In order for this to work in Rails 7+, I would have to add `require_dependency 'admin_user'` to the top of `config/initializers/devise.rb` so that the `AdminUser` class is loaded before evaluating the Devise initializer code, which feels not great. But even if I do so, then Rails will complain `uninitialized constant ApplicationRecord`, so I'd then have to also load the `ApplicationRecord` class, etc. etc. This approach does not allow me to rely on Rails' standard class loading process via `zeitwerk`. So what I'm requesting is for a way to avoid referencing any constants directly in the initializer. I understand that `.classify.constantize` on a string could then present a security risk. Can you think of another approach that would satisfy both requirements (namely not introducing a security risk and not referencing constants that haven't been loaded yet in the initializer)?

**Comment by oivoodoo** [comment]
Hi @mattmenefee 

I've just pushed the changes that you proposed about `constantinize`, let's see if it works for you. Hope it works.

https://github.com/oivoodoo/devise_masquerade/commit/db09650f19e4c6ff1ca17c74fa3fc367aeabb291

thank you!

**Comment by mattmenefee** [comment]
@oivoodoo that worked for me, thanks! Fixed via db09650f19e4c6ff1ca17c74fa3fc367aeabb291.
